### PR TITLE
Expose precompile on registered plugin wrapper

### PR DESCRIPTION
**Pull request by pangratz**
This adds a `precompile` function to the plugin wrapper, so it is possible to precompile template strings within other addons:

```js
// my-addon/index.js
module.exports = {
  name: 'my-addon-using-babel-transformer',

  setupPreprocessorRegistry: function(type, registry) {
    var htmlbarsPlugin = registry.load('template').find(function(plugin) {
      return plugin.name === 'ember-cli-htmlbars';
    });

    // will be a string like 'Ember.HTMLBars.template(function() {...})'
    var precompiled = htmlbarsPlugin.precompile("{{my-component}}");
  }
}
```

---

This addresses pangratz/ember-cli-htmlbars-inline-precompile#7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,3 +39,25 @@ module.exports = {
...
+### Precompile HTMLBars template strings within other addons
+
+```javascript
+module.exports = {
+  name: 'my-addon-name',
+
+  setupPreprocessorRegistry: function(type, registry) {
+    var htmlbarsPlugin = registry.load('template').find(function(plugin) {
+      return plugin.name === 'ember-cli-htmlbars';
+    });
```
Comment (pangratz) [inline README.md]: Is this the way to get the specific wrapper out of the registry?

**Comment by pangratz** [comment]
The test fails due to [`"ember": "components/ember#beta"`](https://github.com/ember-cli/ember-cli-htmlbars/blob/864b02ed739adddc42af27a7f80177043d4e9c1e/bower.json#L4) in `bower.json`. If the `release` channel is used, the tests pass...

File: README.md (hunk 1 above)
Comment (rwjblue) [inline README.md]: Yes, but we should totally add a `registry.findByName` that makes this a simple one liner.  Can be a future thing though.

File: ember-addon-main.js [hunk 2]
```diff
@@ -18,17 +19,21 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    var self = this;
-
     // ensure that broccoli-ember-hbs-template-compiler is not processing hbs files
     registry.remove('template', 'broccoli-ember-hbs-template-compiler');
 
+    var htmlbarsOptions = this.htmlbarsOptions();
```
Comment (rwjblue) [inline ember-addon-main.js]: This needs to be lazy because we need for the full `setupPreprocessorRegistry` phase to be finished before we attempt to "realize" the known plugins (we cannot guarantee that every addon that will register a plugin will have done it before our `setupPreprocessorRegistry`).

tldr; we can't memoize `this.htmlbarsOptions()`, it needs to be called lazily in the `precompile` and `toTree` methods.

**Comment by rwjblue** [comment]
I left one small comment, but otherwise this looks wonderful.

File: README.md (hunk 1 above)
Comment (pangratz) [inline README.md]: :+1:

File: ember-addon-main.js (hunk 2 above)
Comment (pangratz) [inline ember-addon-main.js]: Okidok. Will update.

**Comment by rwjblue** [comment]
Test assertions fixed in https://github.com/ember-cli/ember-cli-htmlbars/pull/54.

**Comment by rwjblue** [comment]
Released in 0.7.7.

**Comment by pangratz** [comment]
Thanks for guiding me through this @rwjblue
